src/frame.js

```diff
@@ -2,30 +2,50 @@ function Frame(frameNumber) {
   this.frameNumber = frameNumber;
   this.remainingPins = 10;
   this.score = 0;
-  this.showScore = true;
   this.complete = false;
   this.bowls = [];
   this.bowlsToDouble = 0;
+
 };
 
 Frame.prototype.bowl = function(x) {
-  console.log("bowling is called on frame " + this.frameNumber)
-
   this.remainingPins -= x;
   this.bowls.push(x);
-  this.score += x
-  console.log("I'm adding a " + x + " to the score")
-
-  if (this.numberOfBowls() == 2 || this.remainingPins == 0) {
-    this.complete = true;
+  this.score += x;
+  if (this.remainingPins == 0)
+  {
+    this.bowlsToDouble = 3 - this.numberOfBowls();
   }
-  if (this.remainingPins == 0) {
-    this.bowlsToDouble = 3 - this.numberOfBowls()
-    console.log("bowls to double = "+this.bowlsToDouble)
+  if(this.frameComplete()) { this.complete = true; }
+};
+
+Frame.prototype.frameComplete = function() {
+  var finalFrameOver = this.frameNumber == 10 && this.bowlsToDouble == 0  && this.numberOfBowls() > 0;
```

Comment:
Nice. I like the way of avoiding lengthy if statement

Reply:
So do I, but I'd prefer splitting that long line to see all the conditions better
```js
var finalFrameOver = (
  this.frameNumber == 10 &&
  this.bowlsToDouble == 0  &&
  this.numberOfBowls() > 0
);
```